### ISSUE TITLE
Wayland Segfault while using discrete NVIDIA GPU

### ISSUE THREAD

**Issue by tcharchar** [RPCS3/rpcs3#15476]
### Quick summary

RPCS3 instantly crashes with a segfault on wayland while using a discrete gpu.

### Details

If `QT_QPA_PLATFORM` is set on `wayland` or `wayland-egl` and any other GPU but the one currently used by wayland is used, RPCS3 instantly crashes and closes with a segfault error.

### Attach a log file

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/15024896/RPCS3.log)

### System configuration

OS: Arch Linux x86_64 
Kernel: 6.8.6-zen1-1-zen 
DE: Hyprland 
CPU: 13th Gen Intel i7-13700H (20) @ 4.800GHz 
GPU: NVIDIA GeForce RTX 4070 Max-Q / Mobile 
GPU: Intel Raptor Lake-P [Iris Xe Graphics] 
Memory:  15713MiB 
NVIDIA Driver: 535.113.01-1, but issue happens on most drivers I've tested from 530.x.x.x and up

### Other details

Changing `QT_QPA_PLATFORM` to `xcb` fixes this issue, however it may be unfavourable to run RPCS3 under x11 due to issues with Xwayland and NVIDIA GPUs as seen [here](https://forums.developer.nvidia.com/t/535-86-05-low-framerate-vulkan-apps-stutter-under-wayland-xwayland/261475).

**Comment by tcharchar** [comment]
From further testing, I can confirm this is a QT issue. [PCSX2 Nightly](https://github.com/PCSX2/pcsx2) (aka pcsx2-qt) also suffers from this, and crashes instantly. However, PCSX2 can actually boot and run dumps while using OpenGL, while RPCS3 hangs with the following:
`QWaylandGLContext::makeCurrent: eglError: 0x300d, this: 0x7228b40083e0 
QWaylandGLContext::makeCurrent: eglError: 0x300d, this: 0x7228b40083e0 
RPCS3: RSX [0x0000000]: SIG: Thread terminated due to fatal error: Could not bind OpenGL context
(in file /usr/src/debug/rpcs3-git/rpcs3/rpcs3/rpcs3qt/gl_gs_frame.cpp:91[:4], in function set_current) (errno=62=Timer expired)`

I should also add, using the `--no-gui` variable still crashes it, because I believe the game window is still being created with the help of QT.

**Comment by Blake96** [comment]
I've found an Android version of RPCS3 emulator on this website: https://apkod.com/ps3-emulator-android/
But I don't see this app on the official GitHub page

**Comment by leucome** [comment]
A similar issue happen with other GPU and specific game.
With a 7900xt on Manjaro with  Plasma 6 Wayland session,  I get segfault  on SoulCalibur V  while other game like Code veronica X is fine. 
Then using  QT_QPA_PLATFORM=xbc  fix SoulCalibur V.   

Linux kernel 6.9.5
Mesa 24.2.0-devel (git-b4bd380704)
KWin version: 6.0.5
Qt Version: 6.7.1
Qt compile version: 6.7.1
XCB compile version: 1.17.0

**Comment by digant73** [comment]
@leucome, @tcharchar is that still an issue?

**Comment by digant73** [comment]
@tcharchar, @leucome please an update on this issue

**Comment by digant73** [comment]
@leucome, @tcharchar please an update on this old issue marked also as external to rpcs3

**Comment by tcharchar** [comment]
I don't believe this issue still happens anymore, but I'd need to test. I have the latest version from git but I'm not sure if it's running in xcb or wayland. I can give a concise answer tomorrow, if that's fine.

**Comment by digant73** [comment]
> I don't believe this issue still happens anymore, but I'd need to test. I have the latest version from git but I'm not sure if it's running in xcb or wayland. I can give a concise answer tomorrow, if that's fine.

sure. waiting for your update after testing

**Comment by tcharchar** [comment]
I'm a bit late again, apologies!

Anyway, no crashes with `QT_QPA_PLATFORM=wayland`. I can also confirm it's no longer an issue since I've apparently been running RPCS3 under wayland for at least two months now. And yes, I have the same specs as above, with the minor difference that I'm on Plasma now. If needed, I can test with Hyprland, or other DEs for that matter.

I also have a (dual gpu) AMD + AMD configuration I can test on, if more info is needed

**Comment by digant73** [comment]
Thanks for checking. I would say you can close this specific ticket. For any other new issue you can always open a new ticket

**Comment by finalfantasian** [comment]
I'm having a similar issue with an AMD GPU and CPU where it crashes with the following on `QT_QPA_PLATFORM=wayland`:
```
wl_display#1: error 0: invalid object 66
The Wayland connection experienced a fatal error: Invalid argument
```

**Comment by ghost** [comment]
> I'm having a similar issue with an AMD GPU and CPU where it crashes with the following on `QT_QPA_PLATFORM=wayland`:
> 
> ```
> wl_display#1: error 0: invalid object 66
> The Wayland connection experienced a fatal error: Invalid argument
> ```

On what linux it crash ?

**Comment by finalfantasian** [comment]
> > I'm having a similar issue with an AMD GPU and CPU where it crashes with the following on `QT_QPA_PLATFORM=wayland`:
> > ```
> > wl_display#1: error 0: invalid object 66
> > The Wayland connection experienced a fatal error: Invalid argument
> > ```
> 
> On what linux it crash ?

Arch Linux on hyprland-git with cachyos repositories enabled.

Edit: Can switch over to standard Arch repos to see if anything changes, btw.
Edit 2: Everything works fine if the above variable is set to `xcb`, btw

**Comment by ghost** [comment]
> > > I'm having a similar issue with an AMD GPU and CPU where it crashes with the following on `QT_QPA_PLATFORM=wayland`:
> > > ```
> > > wl_display#1: error 0: invalid object 66
> > > The Wayland connection experienced a fatal error: Invalid argument
> > > ```
> > 
> > 
> > On what linux it crash ?
> 
> Arch Linux on hyprland-git with cachyos repositories enabled.
> 
> Edit: Can switch over to standard Arch repos to see if anything changes, btw. Edit 2: Everything works fine if the above variable is set to `xcb`, btw

Ok.

**Comment by finalfantasian** [comment]
Update: Even with the official Arch packages it still crashes. Weird. Gives the following:
```
qt.multimedia.ffmpeg: Using Qt multimedia with FFmpeg version n8.1.1 GPL version 3 or later
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
WARNING: radv is not a conformant Vulkan implementation, testing use only.
wl_display#1: error 0: invalid object 47
The Wayland connection experienced a fatal error: Invalid argument
```
When interacting with the menu.

Edit: My bad it was a bug in the compositor I was using that recently received a patch that fixes it.
